test/e2e/set/scyllacluster/scyllacluster_pv.go

```diff
@@ -5,35 +5,272 @@ package scyllacluster
+		const (
+			pvCloneLabelKey = "e2e.operator.scylladb.com/orphaned-pv-test"
+		)
+
 		ctx, cancel := context.WithTimeout(context.Background(), testTimeout)
 		defer cancel()
 
+		nodes, err := f.KubeAdminClient().CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		if len(nodes.Items) != 1 {
+			g.Skip(
+				"This test supports running at most 1 Node, as it relies on the fact that all Pods land on the same Node." +
+					"Replace with actual Node removal logic when CI env is bigger",
+			)
+		}
+
+		rp := corev1.PersistentVolumeReclaimDelete
+		vbm := storagev1.VolumeBindingWaitForFirstConsumer
+		testStorageClass := &storagev1.StorageClass{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "test-",
+			},
+			Provisioner:       "kubernetes.io/no-provisioner",
+			ReclaimPolicy:     &rp,
+			VolumeBindingMode: &vbm,
+		}
+
+		testStorageClass, err = f.KubeAdminClient().StorageV1().StorageClasses().Create(ctx, testStorageClass, metav1.CreateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		defer func() {
+			err := f.KubeAdminClient().StorageV1().StorageClasses().Delete(ctx, testStorageClass.Name, metav1.DeleteOptions{
+				Preconditions: &metav1.Preconditions{
+					UID: &testStorageClass.UID,
+				},
+			})
+			o.Expect(err).NotTo(o.HaveOccurred())
+		}()
+
 		sc := scyllafixture.BasicScyllaCluster.ReadOrFail()
-		sc.Spec.Datacenter.Racks[0].Members = 3
 		sc.Spec.AutomaticOrphanedNodeCleanup = true
+		sc.Spec.Datacenter.Racks[0].Members = 3
+		sc.Spec.Datacenter.Racks[0].Storage.StorageClassName = pointer.String(testStorageClass.Name)
+
+		// This test to trigger the orphaned PV cleanup is updating NodeAffinity of PV used by ScyllaCluster to not exiting Node.
+		// The problem is, that local volume storage provisioners which we use in CI, may set this field, which is immutable.
+		// To overcome this, the ScyllaCluster is using custom StorageClass not backed by any provisioner running in the cluster.
+		// ScyllaCluster is going to request a PVC from that StorageClass, and the test is going to request a clone of the original PVC
+		// from the default StorageClass to get any storage. Then the bound PV is rebounded to the original PVC
+		// but with empty NodeAffinity. This allows the test to trigger the orphaned PV cleanup logic.
+		go func() {
+			defer g.GinkgoRecover()
+
+			lw := &cache.ListWatch{
+				ListFunc: helpers.UncachedListFunc(func(options metav1.ListOptions) (runtime.Object, error) {
+					return f.KubeClient().CoreV1().PersistentVolumeClaims(f.Namespace()).List(ctx, options)
+				}),
+				WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+					return f.KubeClient().CoreV1().PersistentVolumeClaims(f.Namespace()).Watch(ctx, options)
+				},
+			}
+			_, _, watcher, done := watchtools.NewIndexerInformerWatcher(lw, &corev1.PersistentVolumeClaim{})
+			defer func() { <-done }()
+			defer watcher.Stop()
+
+			for {
```

Comment:
what's the difference to UntilWithSync?

Reply:
to use continue instead of `return false, nil` and not caring about timeout error returned when context is cancelled. But changed it anywhay to suggested function.